src/avram/save_operation.cr

```diff
@@ -339,6 +339,37 @@ abstract class Avram::SaveOperation(T)
     !!record_id
   end
 
+  # `#persisted?` always returns `true` in `after_*` hooks, whether
+  # a new record was created, or an existing one was updated.
+  #
+  # This method should always return `true` for a create or `false`
+  # for an update, independent of the stage we are at in the operation.
+  def new_record? : Bool
+    {{ T.resolve.constant(:PRIMARY_KEY_NAME).id }}.value.nil?
```

Comment:
This essentially relies on the fact that we're not currently updating the attributes on an operation after updating a record. We should add a test for this method just in case we decide to change how attributes are handled.

Reply:
Sure, makes a lotta sense.